base/server/upgrade/10.9.0/03-DisableOpenJDKFIPS.py

```diff
@@ -0,0 +1,49 @@
+# Authors:
+#     Alexander Scheel <ascheel@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+# All rights reserved.
+
+from __future__ import absolute_import
+
+import loging
+import os
+import pki
+
+
+class DisableOpenJDKFips(pki.server.upgrade.PKIServerUpgradeScriptlet):
+
+    def __init__(self):
+        super(DisableOpenJDKFips, self).__init__()
+        self.message = 'Set -Dcom.redhat.fips=false in Tomcat configuration'
+
+    def upgrade_instance(self, instance):
+        self.fix_tomcat_config('/etc/pki/%s/tomcat.conf' % instance.name)
+        self.fix_tomcat_config('/etc/sysconfig/%s' % instance.name)
+
+    def fix_tomcat_config(self, filename):
+        if not os.path.exists(filename):
+            logging.debug("Unknown file: %s", filename)
+
+        with open(filename, 'r') as in_fp:
+            lines = in_fp.readlines()
+
+        with open(filename, 'w') as out_fp:
+            for line in lines:
+                if 'JAVA_OPTS="' in line and 'com.redhat.fips' not in line:
```

Comment:
I suppose this will include comments as well, is that OK?

Reply:
Yeah, I think so. If they uncomment it, we'd want them to get a working system, right? And not end up with a broken one and not know what they need to add? My 2c.

Reply:
(the other alternative is adding a comment before this too that says (`# -Dcom.redhat.fips=false not added since JAVA_OPTS uncommented`). I'd take it there was a reason why it was commented out (and it might be broken if `-Dcom.redhat.fips=false` isn't provided _because_ `JAVA_OPTS` was commented out) -- but at least there'd be a path to fixing it.